src/main/java/org/opentripplanner/graph_builder/module/osm/WalkableAreaBuilder.java

```diff
@@ -225,10 +225,12 @@ public void buildWithVisibility(AreaGroup group) {
           continue;
         }
 
-        // Add stops from public transit relations into the area
-        Collection<OSMNode> nodes = osmdb.getStopsInArea(area.parent);
-        if (nodes != null) {
-          for (OSMNode node : nodes) {
+        // Add stops/entrances from public transit relations into the area
+        // they may provide the only entrance to a platform
+        // which otherwise would be pruned as unconnected island
+        Collection<OSMNode> entrances = osmdb.getStopsInArea(area.parent);
```

Comment:
If we change `OsmDatabase` (see above) and return an empty list instead of `null`, we can skip the null check.

Reply:
done